src/components/Modal/BaseModal/BaseModal.tsx

```diff
@@ -1,12 +1,23 @@
-import { useEffect, useRef } from "react";
+import { useEffect, useRef, ReactNode } from "react";
 import styles from "./base.module.css";
 import closeIcon from "assets/images/ic_close.png";
+import { CloseModal } from "../../../utils/hooks/useModal";
 
-function BaseModeal({ title, children, variant, closeModal }) {
-  const modalRef = useRef(null);
+export interface ModalProps {
+  variant: string;
+  closeModal: CloseModal;
+}
+
+interface Props extends ModalProps{
+  title: string;
+  children: ReactNode;
+}
+
+function BaseModeal({ title, children, variant, closeModal }: Props) {
+  const modalRef = useRef<HTMLDivElement>(null);
 
-  const handleOutSideClick = (e) => {
-    if (modalRef.current && !modalRef.current.contains(e.target)) {
+  const handleOutSideClick = (e: MouseEvent) => {
```

Comment:
사소하지만 요렇게 표현되는게 맞을 것 같아요~
```suggestion
  const handleOutsideClick = (e: MouseEvent) => {
```